python/cugraph/cugraph/sampling/node2vec.py

```diff
@@ -109,10 +110,11 @@ def node2vec(G,
         start_vertices = [start_vertices]
 
     if isinstance(start_vertices, list):
-        start_vertices = cudf.Series(start_vertices)
+        # start_vertices = cudf.Series(start_vertices)
+        start_vertices = cudf.Series(start_vertices, dtype='int32')
         if start_vertices.dtype != 'int32':
```

Comment:
Since you just set the dtype to 'int32' in the line above, is this check really needed?

Reply:
For certain types that can be cast to int32 or is int32, like int32, int64, it's not necessary. But I think it'd be better to raise a more specific ValueError like the other checks for max_depth, p, q, compress_result, rather than one from number_map, which would lead the user in a different direction altogether (TypeError: string indices must be integers).